src/main/scala/it/pagopa/interop/backendforfrontend/service/AgreementProcessService.scala

```diff
@@ -25,13 +25,15 @@ trait AgreementProcessService {
   def cloneAgreement(agreementId: UUID)(implicit contexts: Seq[(String, String)]): Future[Agreement]
 
   def getAgreements(
-    producerId: Option[String] = None,
-    consumerId: Option[String] = None,
-    eServiceId: Option[String] = None,
-    descriptorId: Option[String] = None,
-    states: Seq[AgreementState],
-    latest: Option[Boolean] = None
-  )(implicit contexts: Seq[(String, String)]): Future[Seq[Agreement]]
+    producersIds: Seq[UUID] = Seq.empty,
+    consumersIds: Seq[UUID] = Seq.empty,
+    eservicesIds: Seq[UUID] = Seq.empty,
+    descriptorsIds: Seq[UUID] = Seq.empty,
+    states: Seq[AgreementState] = Seq.empty,
+    limit: Int = Int.MaxValue,
```

Comment:
this is a bit dangerous, and it will be rejected by the agreement process because its max allowed value is 50.
Lets remove this default